plugins/hazelcast3/src/main/java/org/radargun/service/HazelcastQueryable.java

```diff
@@ -206,6 +238,70 @@ public QueryResult execute() {
       }
    }
 
+   private class HazelcastAggregationQuery implements Query {
+      private final IMap map;
+      private final PropertyExtractor propertyExtractor;
+      Aggregation aggregation;
+      Supplier supplier;
+
+      public HazelcastAggregationQuery(IMap map, Class<?> clazz, Predicate predicate, SelectExpression aggregatedAttribute) {
+         this.map = map;
+         Accessor accessor = getAccessor(clazz, aggregatedAttribute.attribute);
+         propertyExtractor = new ReflexivePropertyExtractor(accessor);
+
+         Supplier reflexiveSupplier = Supplier.all(propertyExtractor);
+         if (predicate != null) {
+            supplier = Supplier.fromPredicate(predicate, reflexiveSupplier);
+         } else {
+            supplier = reflexiveSupplier;
+         }
+
+         AggregationFunction function = aggregatedAttribute.function;
+         String type = accessor.getReturnType().getSimpleName();
+
+         aggregation = null;
+         if (function == AggregationFunction.NONE)
```

Comment:
Two-level switch would look better here, IMO; though I am not sure if switch works on classes (rather not, I suppose).